components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/APIMappingUtil.java

```diff
@@ -3051,4 +2935,43 @@ private static Date parseStringToDate(String time) throws java.text.ParseExcepti
         SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
         return dateFormat.parse(time);
     }
+
+    public static APIEndpointListDTO fromAPIEndpointListToDTO(List<APIEndpointInfo> apiEndpoints)
+            throws APIManagementException {
+        APIEndpointListDTO apiEndpointListDTO = new APIEndpointListDTO();
+        List<APIEndpointDTO> apiEndpointDTOs = new ArrayList<>();
+        for (APIEndpointInfo apiEndpoint : apiEndpoints) {
+            apiEndpointDTOs.add(fromAPIEndpointToDTO(apiEndpoint));
+        }
+        apiEndpointListDTO.setCount(apiEndpointDTOs.size());
+        apiEndpointListDTO.setList(apiEndpointDTOs);
+        return apiEndpointListDTO;
+    }
+
+    public static APIEndpointDTO fromAPIEndpointToDTO(APIEndpointInfo apiEndpoint) throws APIManagementException {
+        APIEndpointDTO apiEndpointDTO = new APIEndpointDTO();
+        apiEndpointDTO.setId(apiEndpoint.getEndpointUuid());
+        apiEndpointDTO.setName(apiEndpoint.getEndpointName());
+        apiEndpointDTO.setEndpointType(apiEndpoint.getEndpointType());
+        apiEndpointDTO.setEndpointConfig(apiEndpoint.getEndpointConfig());
+        return apiEndpointDTO;
+    }
+
+    public static APIEndpointInfo fromDTOtoAPIEndpoint(APIEndpointDTO apiEndpointDTO, String organization)
+            throws APIManagementException {
+        APIEndpointInfo apiEndpoint = new APIEndpointInfo();
+        apiEndpoint.setEndpointUuid(apiEndpointDTO.getId());
+        apiEndpoint.setEndpointName(apiEndpointDTO.getName());
+        apiEndpoint.setEndpointType(apiEndpointDTO.getEndpointType());
+        try {
+            HashMap endpointConfigHashMap = (HashMap) apiEndpointDTO.getEndpointConfig();
+            apiEndpoint.setEndpointConfig(endpointConfigHashMap);
+        } catch (ClassCastException e) {
+            throw new APIManagementException("Endpoint Config is missing of API Endpoint.",
```

Comment:
ClassCastException is a runtime exception which cannot be detected at compile time. instead of catching it we can try to handle it differently. Please check https://rollbar.com/blog/java-classcastexception/

Reply:
I previously used instanceOf for the class cast but it always becomes false o it always set null to endpoint config.
But if I do like this it works fine, it means it tries to convert to hashmap.  However, I need a downcast to hashMap for this Object.